term/src/terminalstate/image.rs

```diff
@@ -184,6 +184,75 @@ impl TerminalState {
+                 *
+                 *   - Fully opaque image (all pixels alpha = 255)
+                 *       Kitty       => cell.attrs_mut().attach_image(img)
+                 *       Sixel/Iterm => cell.attrs_mut().set_image(img)
+                 *
+                 *   - Partially transparent image (neither of the
+                 *     above two cases)
+                 *       Kitty       => cell.attrs_mut().attach_image(img)
+                 *       Sixel/Iterm => cell.attrs_mut().blit_over_image(img)
+                 *
+                 * This way fully-invisible images are culled
+                 * immediately (performance boost for all image
+                 * styles), and sixel/iTerm2 blend on top as expected
+                 * in other terminals.
+                 *
+                 * The nice thing is that the check for partially
+                 * transparent/fully transparent/opaque can occur in a
+                 * single pass through the pixels, like:
+                 *
+                 *    let mut all_transparent = false;
+                 *    let mut all_opaque = false;
+                 *    let mut mixed = false;
+                 *    foreach pixel {
+                 *      // If any pixel is in-between, we are done.
+                 *      if pixel.alpha > 0 && pixel.alpha < 255 {
+                 *        mixed = true;
+                 *        break;
+                 *      }
+                 *
+                 *      // Unfortunately if they are all one-way or the other
+                 *      // the check has to go to the end.
+                 *      if pixel.alpha > 0 all_transparent = false;
+                 *      if pixel.alpha < 255 all_opaque = false;
+                 *    }
+                 *
+                 *    // There should be only one of the booleans set
+                 *    // now if mixed is false.
+                 *    assert (!mixed && (all_transparent != all_opaque))
+                 *            || mixed;
+                 *
+                 *    if all_transparent => skip this cell;
+                 *    else if Kitty => do the same as now
+                 *    else if all_opaque => do the same as now
+                 *    else cell.attrs_mut().blit_over_image(img)
+                 */
+
+                /*
+                // The end of the proper Rust code might look like this:
+                match (params.style, translucence) {
+                    (_, Transparent) => {},
+                    (ImageAttachStyle::Kitty, _) => cell.attrs_mut().attach_image(img),
+                    (ImageAttachStyle::Sixel | ImageAttachStyle::Iterm, Opaque) => {
+                        cell.attrs_mut().set_image(img)
+                    },
+                    (ImageAttachStyle::Sixel | ImageAttachStyle::Iterm, Mixed) => {
+                        cell.attrs_mut().blit_over_image(img)
+                    },
+                };
+                */
```

Comment:
@wez Here is where I am stuck.  (And gonna press the "Add single comment" button rather than "Start a review"... Sorry still don't know GH that well.)